components/global_search/global-search.react.js

```diff
@@ -0,0 +1,162 @@
+import React, {Component} from 'react';
+import './style.scss';
+import SearchOption from "./child_components/search_option/SearchOption";
+
+class GlobalSearch extends Component {
+
+    constructor(props) {
+        super(props);
+
+        this.show_search_options_ref = React.createRef();
+        this.show_hide_mobile_ref = React.createRef();
+
+        this.state = {
+
+            show_search_options: false,
+            show_mobile_search: false,
+
+            active_search: {}, // This is assigned to the first option below
+
+            search_selector: {
+                label: 'Show search options',
+                id: 'show_options',
+            },
+
+            component_label: 'Search our website or catalogue',
+            search_query_legend: 'Enter search term',
+            search_options: {
+                group_name: 'search_type',
+                select_type: 'Select a search type',
+                options: [
+                    {
+                        label: 'Search our website',
+                        id: 'website_search',
+                        search_id: 'website_search_box',
+                        url: 'https://www.nationalarchives.gov.uk/search/results'
+                    },
+                    {
+                        label: 'Search our records',
+                        id: 'discovery_search',
+                        search_id: 'discovery_search_box',
+                        url: 'https://discovery.nationalarchives.gov.uk/results/r'
+                    }
+                ]
+            }
+        };
+
+        const {state} = this;
+        state.active_search = state.search_options.options[0];
+
+    }
+
+    handle_search_selection = (e) => {
+        if (e.target.type === 'radio') {
+            let selection = this.state.search_options.options.find((i) => {
+                return (e.target.id === i.id);
+            });
+            this.setState({active_search: selection});
+        }
+    }
+
+    toggle_search_options = () => {
+        this.setState({show_search_options: !this.state.show_search_options})
```

Comment:
Again same as above
You don't need to use `this.state` as you are destructuring `this` on line `47`